projects/storefrontlib/src/lib/store-finder/components/store-finder-grid/store-finder-grid.component.ts

```diff
@@ -44,27 +45,29 @@ export class StoreFinderGridComponent implements OnInit {
           locations.pointOfServices &&
           locations.pointOfServices.length === 1
         ) {
-          this.router.navigate([
-            'store-finder',
-            'country',
-            this.route.snapshot.params.country,
-            'region',
-            this.route.snapshot.params.region,
-            locations.pointOfServices[0].name
-          ]);
+          this.viewStore(locations.pointOfServices[0]);
         }
         this.locations = locations;
       });
   }
 
   viewStore(location: any): void {
-    this.router.navigate([
-      'store-finder',
-      'country',
-      this.route.snapshot.params.country,
-      'region',
-      this.route.snapshot.params.region,
-      location.name
-    ]);
+    if (this.route.snapshot.params.region) {
```

Comment:
Maybe we should use destructuring and get rid of `else` statement?
```
const { region, country } = this.route.snapshot.params;
if (region) {
     this.router.navigate(....)
     return;
}
this.router.navigate(...)
```

Or better
```
const { region, country } = this.route.snapshot.params;
const params = [
   'store-finder',
   'country',
   country,
];
if (region) {
    params.push( 'region', region, location.name );
}
this.router.navigate(params);
```